src/importexport/mei/internal/meiconverter.cpp

```diff
@@ -3333,6 +3357,52 @@ std::list<std::string> Convert::getTypeValuesWithPrefix(const std::string& typeS
     return values;
 }
 
+void Convert::layerIdentFromMEI(engraving::EngravingItem* item, const libmei::Element& meiElement)
+{
+    const libmei::AttLayerIdent* layerAtt = dynamic_cast<const libmei::AttLayerIdent*>(&meiElement);
+
+    IF_ASSERT_FAILED(layerAtt) {
+        return;
+    }
+
+    // @color
+    if (layerAtt->HasLayer()) {
+        // without further check we assume the layer to match
+        if (item->hasVoiceAssignmentProperties()) {
```

Comment:
It might make sense to do `if (!item->hasVoiceAssignmentProperties()) { return; }` directly at the top of the method; if this is false, then the element won't accept these properties anyway so there is no need to look for them.